src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -249,11 +284,9 @@ public SafeDeleteContext(SafeFreeCredentials credential, bool isServer, bool rem
             }
             catch(Exception ex)
             {
-                if (gotCredRef)
-                {
-                    _credential.DangerousRelease();
-                }
                 Debug.Write("Exception Caught. - " + ex);
+                Debug.Assert((null != credential), "Null credential in SafeDeleteContext");
+                Dispose();
```

Comment:
Calling Dispose here is right, but in Dispose implementation below you're calling _sslContext.Dispose, and if the ctor throws in the call to AllocateSslContext, _sslContext will be null, and the call to Dispose will dereference null.  You need to check in Dispose that _sslContext is not null before disposing it.  Then you should set it back to null.